From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Wed, 23 Nov 2022 21:03:04 +0100
Subject: [PATCH] Configuration - Get Fiddle configuration with Bukkit API

Allows getting the Fiddle configuration using the Bukkit API.
This is done by providing the abstract method `Server.getFiddleConfig()`, which is implemented in the "Configuration - Get Fiddle configuration with Bukkit API" patch in the `fiddle-server` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

This patch is based on the following patch:
"Gale configuration"
By: Martijn Muijsers <martijnmuijsers@live.nl>
As part of: Gale (https://github.com/GaleMC/Gale)
Licensed under: MIT (https://opensource.org/licenses/MIT)

* Gale description *

License: MIT (https://opensource.org/licenses/MIT)
Gale - https://galemc.org

This patch is based on the following patch:
"Timings v2"
By: Aikar <aikar@aikar.co>
As part of: Paper (https://github.com/PaperMC/Paper)
Licensed under: MIT (https://opensource.org/licenses/MIT)

diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 9af4bc16da09e59009c47911219e99450cdf2aa5..6837162b01b208955973b70a694ef4e69fcedc20 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2196,6 +2196,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         {
             throw new UnsupportedOperationException("Not supported yet.");
         }
+
+        // Fiddle start - Fiddle configuration - API
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getFiddleConfig()
+        {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Fiddle end - Fiddle configuration - API
+
         // Paper end
 
         /**
